service/javadsl/integration-client/src/main/java/com/lightbend/lagom/javadsl/client/integration/LagomClientFactory.java

```diff
@@ -170,90 +177,198 @@ public void close() {
+        return Environment.apply(new File("."), classLoader, Mode.Prod$.MODULE$);
+    }
+
+    private static Config buildConfig(ClassLoader classLoader) {
+        return Configuration.load(
+            classLoader,
+            System.getProperties(),
+            Map$.MODULE$.empty(),
+            true
+        ).underlying();
+    }
+    /**
+     * Creates a Lagom client factory.
+     *
+     * Generally, there should be only one instance of this per system.  Additionally, since it holds thread and connection
+     * pools, it must be shutdown, by calling the {@link #close()} method, when no longer needed.
+     *
+     * This method should be used whenever your application has already a running ActorSystem. In that case it's preferable to reuse it
+     * inside LagomClientFactory instead to let the factory create and manage its own (see {@link #create(String, ClassLoader)}).
+     *
+     * Calling {@link #close()} on a {@link LagomClientFactory} created using this method will NOT terminated
+     * the passed {@link ActorSystem} and {@link Materializer}.
+     *
+     * @param serviceName The name of this service that is going to be accessing the services created by this factory.
+     * @param classLoader The classloader.
+     * @param actorSystem An existing ActorSystem
+     * @param materializer An ActorMaterializer
+     * @return The client factory.
+     */
+    public static LagomClientFactory create(String serviceName, ClassLoader classLoader, ActorSystem actorSystem, Materializer materializer) {
+        return create(
+            serviceName,
+            buildtEnvironment(classLoader),
+            buildConfig(classLoader),
+            actorSystem,
+            materializer,
+            /*managedActorSystem*/ false);
+    }
+
+
+    /**
+     * Creates a Lagom client factory.
+     *
+     * Generally, there should be only one instance of this per system.  Additionally, since it holds thread and connection
+     * pools, it must be shutdown, by calling the {@link #close()} method, when no longer needed.
+     *
+     * This method should be used whenever your application does NOT have a running {@link ActorSystem}
+     * and you don't want to manage one yourself.
+     *
+     * Internally, this method will create a new {@link ActorSystem} that will be attached to the lifecycle of this factory.
+     * In other words, the internal {@link ActorSystem} will terminate upon calling {@link #close()}.
+     *
+     * In case your application already have a running {@link ActorSystem}, we recommend
+     * to use {@link #create(String, ClassLoader, ActorSystem, Materializer)}) instead.
+     *
+     * @param serviceName The name of this service that is going to be accessing the services created by this factory.
+     * @param classLoader The classloader.
+     * @return The client factory.
+     */
+    public static LagomClientFactory create(String serviceName, ClassLoader classLoader) {
```

Comment:
I was also in doubt if we should deprecate this method and only promote the one that receives an ActorSystem. 

Creating an ActorSystem on users' behalf is probably not the best thing to do. 

If we compare this with the upcoming akka-grpc, when we create a client we must provide an `ActorSystem` and `Materializer`. The client is not responsible to create and manage one for the users. That's the most correct pattern and we should probably adopt it for the LagomClientFactory as well.

WDYT?